test/hierarchical/test_ordinal.py

```diff
@@ -0,0 +1,310 @@
+    importer = pypesto.petab.PetabImporter(petab_problem, ordinal=True)
+    importer.create_model()
+
+    objective = importer.create_objective(
+        inner_solver_options=option,
+    )
+    problem = importer.create_problem(objective)
+    return problem
+
+
+def test_optimal_scaling_calculator_and_objective():
+    """Test the optimal scaling calculation of objective values and."""
+    petab_problem = petab.Problem.from_yaml(example_ordinal_yaml)
+
+    methods = [STANDARD, REDUCED]
+
+    options_per_method = {
+        STANDARD: {'method': STANDARD, 'reparameterized': False},
+        REDUCED: {'method': REDUCED, 'reparameterized': False},
+    }
+    problems = {}
+
+    for method, options in options_per_method.items():
+        importer = pypesto.petab.PetabImporter(petab_problem, ordinal=True)
+        objective = importer.create_objective(
+            inner_solver_options=options,
+        )
+        problem = importer.create_problem(objective)
+        problems[method] = problem
+
+    def calculate(problem, x_dct):
+        return problem.objective.calculator(
+            x_dct=x_dct,
+            sensi_orders=(0, 1),
+            mode=MODE_FUN,
+            amici_model=problem.objective.amici_model,
+            amici_solver=problem.objective.amici_solver,
+            edatas=problem.objective.edatas,
+            n_threads=1,
+            x_ids=petab_problem.x_ids,
+            parameter_mapping=problem.objective.parameter_mapping,
+            fim_for_hess=False,
+        )
+
+    x_dct = dict(zip(petab_problem.x_ids, petab_problem.x_nominal_scaled))
+
+    calculator_results = {
+        method: calculate(problems[method], x_dct=x_dct) for method in methods
+    }
+
+    # The results of the objective gradient and function value
+    # should not depend on the method given.
+    assert np.isclose(
+        calculator_results[STANDARD]['fval'],
+        calculator_results[REDUCED]['fval'],
+    )
+    assert np.isclose(
+        calculator_results[STANDARD]['grad'],
+        calculator_results[REDUCED]['grad'],
+    ).all()
+
+    # Since the nominal parameters are close to true ones, the
+    # the fval and grad should both be low.
+    assert np.all(calculator_results[STANDARD]['fval'] < 0.2)
+    assert np.all(calculator_results[STANDARD]['grad'] < 0.1)
+
+
+def _inner_problem_exp():
+    timepoints = np.linspace(1, 10, 10)
+
+    expected_inner_parameter_values = {
+        STANDARD: np.asarray(
+            [
+                2.07281338e-14,
+                1.99999999e00,
+                2.58823529e00,
+                4.00689557e00,
+                4.59542030e00,
+                6.23751538e00,
+                6.82578436e00,
+                8.40233862e00,
+                8.99060060e00,
+                1.10767572e01,
+            ]
+        ),
+        REDUCED: np.asarray(
+            [1.99999999, 4.00689557, 6.23751538, 8.40233862, 11.07675724]
```

Comment:
Yeah true. They should be the same. The true value is 11.07675723985916

What happened is that when I printed one and the other, due to the different "notation" of printing, for one case it printed one decimal more. 
In any case, doesn't matter as much since I selected `rtol=1e-3`. But will change for correctness.